File: benchmark/corpus/java/cases/vuln_sqli.java
import java.sql.Connection;
import java.sql.SQLException;
import java.sql.Statement;
import javax.servlet.http.HttpServletRequest;

public class VulnSqli {
    void handle(Connection conn, HttpServletRequest request) throws SQLException {
        String id = request.getParameter("id");
        Statement stmt = conn.createStatement();
        // ZS-JAVA-001: SQL injection — query built by concatenating tainted id
        stmt.executeQuery("SELECT * FROM users WHERE id = " + id);
    }
}

File: benchmark/corpus/php/cases/vuln_sqli.php
<?php
// ZS-PHP-002: SQL injection — query built by concatenating tainted $id
$conn = mysqli_connect("localhost", "user", "pass", "db");
$id = $_GET['id'];
$query = "SELECT * FROM users WHERE id = " . $id;
mysqli_query($conn, $query);
